src/infrastructure/transport/notes-api/index.ts

```diff
@@ -0,0 +1,40 @@
+import Transport from '../index';
+import ApiResponse from './types/ApiResponse';
+
+/**
+ * Notes api transport
+ */
+export default class NotesApiTransport extends Transport {
+  /**
+   * Constructor for notes api transport
+   *
+   * @param baseUrl - Base URL
+   */
+  constructor(baseUrl: string) {
+    super(baseUrl);
+  }
+
+  /**
+   * Make GET request
+   *
+   * @template Payload - response payload type
+   * @param endpoint - API endpoint
+   * @returns { Promise<Payload | null> } - response payload
+   */
+  public async get<Payload>(endpoint: string): Promise<Payload | null> {
+    const response = await super.get<ApiResponse<Payload>>(endpoint);
+
+    /**
+     * If error in response
+     */
+    if (response.status) {
```

Comment:
its not obvious that `status` can be only in an error

Reply:
for now i am checking for data